eng/native/configuretools.cmake

```diff
@@ -39,19 +39,21 @@ if(NOT WIN32)
     set(${var} ${EXEC_LOCATION_${exec}} PARENT_SCOPE)
   endfunction()
 
-  locate_toolchain_exec(ar CMAKE_AR)
-  locate_toolchain_exec(link CMAKE_LINKER)
-  locate_toolchain_exec(nm CMAKE_NM)
+  if(NOT CLR_CMAKE_TARGET_ANDROID OR CROSS_ROOTFS)
+    locate_toolchain_exec(ar CMAKE_AR)
+    locate_toolchain_exec(link CMAKE_LINKER)
+    locate_toolchain_exec(nm CMAKE_NM)
+  endif()
 
   if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     locate_toolchain_exec(ranlib CMAKE_RANLIB)
   endif()
 
-  if(NOT CLR_CMAKE_TARGET_DARWIN AND NOT CLR_CMAKE_TARGET_IOS)
+  if(NOT CLR_CMAKE_TARGET_DARWIN AND NOT CLR_CMAKE_TARGET_IOS AND (NOT CLR_CMAKE_TARGET_ANDROID OR CROSS_ROOTFS))
     locate_toolchain_exec(objdump CMAKE_OBJDUMP)
 
-    if(CMAKE_SYSTEM_NAME STREQUAL Android)
-      set(TOOSET_PREFIX ${ANDROID_TOOLCHAIN_PREFIX})
+    if(CLR_CMAKE_TARGET_ANDROID)
+      set(TOOLSET_PREFIX ${ANDROID_TOOLCHAIN_PREFIX})
```

Comment:
@am11 this one had a typo in it so I'm not sure how it worked for you.

Reply:
@akoeplinger, NDK has both with and without prefix variants that's how I think. You can try out this docker https://github.com/dotnet/runtime/issues/33237#issuecomment-601054590.